backend/utils/utils.py

```diff
@@ -53,16 +55,23 @@ def extract_token_from_auth_header(auth_header: str):
     return auth_header[len("Bearer ") :]
 
 
-def verify_token(request):
-    try:
-        bearer = request.headers["authorization"]
-        if bearer:
-            token = bearer[len("Bearer ") :]
-            decoded = jwt.decode(
-                token, JWT_SECRET_KEY, options={"verify_signature": False}
+def verify_auth_token(auth_token: HTTPAuthorizationCredentials = Depends(HTTPBearer())):
+    data = decode_token(auth_token.credentials)
+    if data != None and "email" in data:
+        user = Users.get_user_by_email(data["email"])
+        if user is None:
+            raise HTTPException(
+                status_code=status.HTTP_401_UNAUTHORIZED,
+                detail=ERROR_MESSAGES.INVALID_TOKEN,
             )
-            return decoded
-        else:
-            return None
-    except Exception as e:
-        return None
+        return
+    else:
+        raise HTTPException(
+            status_code=status.HTTP_401_UNAUTHORIZED,
+            detail=ERROR_MESSAGES.UNAUTHORIZED,
+        )
+
+
+def get_current_user(auth_token: HTTPAuthorizationCredentials = Depends(HTTPBearer())):
+    data = decode_token(auth_token.credentials)
+    return Users.get_user_by_email(data["email"])
```

Comment:
I believe it would be better to merge `verify_auth_token` and `get_current_user`, so that we don't need additional dependencies for `/chat`, `/users`, and `/modelfiles` api routes. Any reasoning behind the separation of the two? Seems like to me that there isn't any reason for the backend to make the same user query to db twice. Here's my suggestion:


```suggestion
def get_current_user(auth_token: HTTPAuthorizationCredentials = Depends(HTTPBearer())):
    data = decode_token(auth_token.credentials)

    if data != None and "email" in data:
        user = Users.get_user_by_email(data["email"])
        if user:
            return user
        else:
            raise HTTPException(
                status_code=status.HTTP_401_UNAUTHORIZED,
                detail=ERROR_MESSAGES.INVALID_TOKEN,
            )
    else:
        raise HTTPException(
            status_code=status.HTTP_401_UNAUTHORIZED,
            detail=ERROR_MESSAGES.UNAUTHORIZED,
        )
```

Thanks!

Reply:
```python
app.include_router(
    users.router,
    prefix="/users",
    tags=["users"],
    dependencies=[Depends(get_current_user)],
)
```

If I do the above and remove the second function, then in my actual route I don't have access to the user. I would have to refetch it anyways, as `Depends` doesn't inject the value into `request` object. We could do `request.state.user = user # fetch user from db` in the util function, but that becomes implicit. I don't think a second call to DB to fetch the user by id is going to affect much when everything is local.

Reply:
I meant getting rid of `dependencies` like such:

```python
app.include_router(
    users.router,
    prefix="/users",
    tags=["users"]
)
```

and leaving the router functions as they are!

Reply:
Hmm, I thought having token validation at the base route as above would be better and less copy-paste of the same dependency on users.router when adding new endpoints. I can make the change as suggested in the original comment.

Reply:
Yep, I believe that would be best, as we don't know what we'll be adding to those api routes in the future 😅 We can always come back to this and adopt your approach if necessary, Thanks!

Reply:
Pushed the changes as discussed.